src/Utilities/AnsiBuilder.cs

```diff
@@ -0,0 +1,330 @@
+            var newChars = new char[charsCapacity];
+            buffer.CopyTo(newChars, 0);
+            buffer = newChars;
+        }
+
+        /// <summary>Parses a string for special tags and replaces them with ANSI codes.</summary>
+        private void Parse()
+        {
+            // Seek out all tags and replace them with the equivalent ANSI codes.
+            // Always scan for the most common attributes ('normal', 'new line' and foreground color
+            // codes) for replacements, but after that we will be more selective about how we spend
+            // our time with priority-ordered replacement batches while tokens are still present.
+            Replace("<%n%>", AnsiSequences.TextNormal);
+            Replace("<%nl%>", AnsiSequences.NewLine);
+            Replace("<%black%>", AnsiSequences.ForegroundBlack);
+            Replace("<%red%>", AnsiSequences.ForegroundRed);
+            Replace("<%green%>", AnsiSequences.ForegroundGreen);
+            Replace("<%yellow%>", AnsiSequences.ForegroundYellow);
+            Replace("<%blue%>", AnsiSequences.ForegroundBlue);
+            Replace("<%magenta%>", AnsiSequences.ForegroundMagenta);
+            Replace("<%cyan%>", AnsiSequences.ForegroundCyan);
+            Replace("<%white%>", AnsiSequences.ForegroundWhite);
+
+            // Background Colors: only scan for these if we have any "<%b" left, which 
+            // most likely indicates a background color setting.
+            Replace("<%bblack%>", AnsiSequences.BackgroundBlack);
+            Replace("<%bred%>", AnsiSequences.BackgroundRed);
+            Replace("<%bgreen%>", AnsiSequences.BackgroundGreen);
+            Replace("<%byellow%>", AnsiSequences.BackgroundYellow);
+            Replace("<%bblue%>", AnsiSequences.BackgroundBlue);
+            Replace("<%bmagenta%>", AnsiSequences.BackgroundMagenta);
+            Replace("<%bcyan%>", AnsiSequences.BackgroundCyan);
+            Replace("<%bwhite%>", AnsiSequences.BackgroundWhite);
+
+
+            Replace("<%reset%>", AnsiSequences.TextNormal); // Avoid usage; favor the <%n%> form.
+            Replace("<%b%>", AnsiSequences.TextBold);
+            Replace("<%cls%>", AnsiSequences.ClearScreenAndHomeCursor);
+            Replace("<%u%>", AnsiSequences.TextUnderline);
+            Replace("<%underline%>", AnsiSequences.TextUnderline);
+            Replace("<%hidden%>", AnsiSequences.TextHidden);
+            
+            // Cursor Movement
+            Replace("<%up%>", AnsiSequences.MoveCursorUp1);
+            Replace("<%down%>", AnsiSequences.MoveCursorDown1);
+            Replace("<%left%>", AnsiSequences.MoveCursorLeft1);
+            Replace("<%right%>", AnsiSequences.MoveCursorRight1);
+
+            // MXP Bits
+            // Regex to match <%mxp <!I can be an mxp tag>%> is <\%mxp (.|\n)+?%>
+            Replace("<%mxpopenline%>", AnsiSequences.MxpOpenLine);
+            Replace("<%mxpsecureline%>", AnsiSequences.MxpSecureLine);
```

Comment:
If we end up keeping the token replacements here, we need to remember to remove these ones.  (Especially MXP Secure Lines need their own secure API per #107.)

Reply:
I agree, just figured id leave them here until a decision on how to handle them was done.